hvplot/plotting/scatter_matrix.py

```diff
@@ -38,19 +49,50 @@ def scatter_matrix(data, c=None, chart='scatter', diagonal='hist', alpha=0.5, **
     diagonal = HoloViewsConverter._kind_mapping[diagonal]
     chart = HoloViewsConverter._kind_mapping[chart]
 
+    #remove datashade kwds
+    if datashade == True or rasterize == True:
+        ds_kwds = {}
+        if 'aggregator' in kwds:
+            ds_kwds['aggregator'] = kwds.pop('aggregator')
+        
+    #remove dynspread kwds
+    if dynspread == True:
+        dn_kwds = {}
+        if 'max_px' in kwds:
+            dn_kwds['max_px'] = kwds.pop('max_px')
+        if 'threshold' in kwds:
+            dn_kwds['threshold'] = kwds.pop('threshold')
+
     colors = _hv.plotting.util.process_cmap('Category10', categorical=True)
     chart_opts = dict(alpha=alpha, cmap=colors, tools=['box_select', 'lasso_select'],
                       nonselection_alpha=0.1, **kwds)
 
     grid = _hv.operation.gridmatrix(data, diagonal_type=diagonal, chart_type=chart)
+
     if c:
         chart_opts['color_index'] = c
         grid = grid.map(lambda x: x.clone(vdims=x.vdims+[c]), 'Scatter')
         groups = _hv.operation.gridmatrix(data.groupby(c).overlay(),
                                           chart_type=chart,
                                           diagonal_type=diagonal)
-        grid = (grid * groups).map(lambda x: x.get(0) if isinstance(x.get(0), chart) else x.get(1),
+        if datashade == True:
+            grid = (grid * groups).map(lambda x: x.get(0).apply(_datashade, **ds_kwds) if isinstance(x.get(0), chart) else x.get(1),
+                                   _hv.Overlay)
+        elif dynspread == True:
+            grid = (grid * groups).map(lambda x: x.get(0)
+            .apply(_datashade, **ds_kwds).
+            apply(_dynspread, **dn_kwds) if isinstance(x.get(0), chart) else x.get(1),
```

Comment:
Splitting `.` and `apply` here seems very confusing.

Reply:
that is an oops.  I fixed that and didn't push it.  I don't like that either.